apps/andi/lib/andi_web/live/ingestion_live_view/metadata/metadata_form.ex

```diff
@@ -1,209 +1,112 @@
+        </div>
+
+        <div class="ingestion-metadata-form ingestion-metadata-form__target-dataset">
+          <%= label(f, :targetDatasetName, "Dataset Name", class: "label label--required") %>
+          <%= hidden_input(f, :targetDataset, value: selected_dataset) %>
+          <%= text_input(f, :targetDatasetName, [class: "input ingestion-form-fields", value: get_dataset_name(selected_dataset), disabled: true, required: true]) %>
+          <button class="btn btn--select-dataset-search btn--primary-outline" phx-click="select-dataset" phx-target="<%= @myself %>" type="button">Select Dataset</button>
+          <%= ErrorHelpers.error_tag(f, :targetDataset, bind_to_input: false) %>
+        </div>
+      </form>
+      <%= live_component(@socket, AndiWeb.IngestionLiveView.SelectDatasetModal,
+            selected_dataset: selected_dataset,
+            visibility: @select_dataset_modal_visibility,
+            id: :ingestion_metadata_search,
+            close_modal_callback: &close_modal/0
+      ) %>
+    </div>
     """
   end
 
-  def handle_info(
-        %{topic: "form-save", event: "save-all", payload: %{ingestion_id: ingestion_id}},
-        %{assigns: %{changeset: changeset}} = socket
-      ) do
-    original_ingestion = Ingestions.get(ingestion_id)
-    {status, _} = Ingestions.update(original_ingestion, changeset.changes)
-    valid? = if status == :ok, do: "valid", else: "invalid"
-    FormUpdate.send_value(socket.parent_pid, {:update_save_message, valid?})
-    {:noreply, socket}
-  end
-
-  def handle_info(
-        %{topic: "ingestion-published"},
-        socket
-      ) do
-    {:noreply, assign(socket, ingestion_published?: true)}
-  end
-
   def handle_event("select-dataset", _, socket) do
     {:noreply,
      assign(socket,
-       select_dataset_modal_visibility: "visible",
-       old_selected_dataset: socket.assigns.selected_dataset
+       select_dataset_modal_visibility: "visible"
      )}
   end
 
-  def handle_event("select-dataset-search", %{"id" => id}, socket) do
-    if(socket.assigns.selected_dataset == id) do
-      {:noreply, assign(socket, selected_dataset: nil, old_selected_dataset: id)}
-    else
-      {:noreply, assign(socket, selected_dataset: id, old_selected_dataset: socket.assigns.selected_dataset)}
-    end
-  end
-
-  def handle_event("remove-selected-dataset", %{"id" => id}, socket) do
-    {:noreply, assign(socket, selected_dataset: nil, old_selected_dataset: id)}
-  end
-
-  def handle_event("validate", %{"form_data" => form_data, "_target" => ["form_data", "sourceFormat"]}, socket) do
-    AndiWeb.Endpoint.broadcast_from(self(), "source-format", "format-update", %{
-      new_format: form_data["sourceFormat"],
-      ingestion_id: socket.assigns.ingestion_id
-    })
-
-    form_data
-    |> IngestionMetadataFormSchema.changeset_from_form_data()
-    |> complete_validation(socket)
-  end
-
   def handle_event("validate", %{"form_data" => form_data}, socket) do
```

Comment:
Since all the metadata should be validated on the API/top level, we send a message to our parent with our updated IngestionMetadataFormSchema changeset but keep our socket the same because our parent will tell us what to render